adapters/repos/db/vector/hnsw/search.go

```diff
@@ -237,13 +254,15 @@ func (h *hnsw) searchLayerByVectorWithDistancer(ctx context.Context,
 		return nil, errors.Wrapf(err, "calculate distance of current last result")
 	}
 	var connectionsReusable []uint64
-
 	var sliceConnectionsReusable *common.VectorUint64Slice
 	var slicePendingNextRound *common.VectorUint64Slice
 	var slicePendingThisRound *common.VectorUint64Slice
 
-	if allowList != nil && useAcorn {
-		sliceConnectionsReusable = h.pools.tempVectorsUint64.Get(M * h.maximumConnectionsLayerZero)
+	if allowList == nil {
+		strategy = SWEEPING
+	}
+	if strategy == ACORN {
+		sliceConnectionsReusable = h.pools.tempVectorsUint64.Get(8 * h.maximumConnectionsLayerZero)
```

Comment:
Do you think `8 * maximumConnectionsLayerZero` (so normally 8*64) should be a dynamic parameter users can set?

Reply:
Maybe, but it will most likely be a dead branch on the code after we merge the full re-entries logic. What do you think? We already have some good results there so maybe we do not even need ACORN after that.